components/LessonLocked.tsx

```diff
@@ -0,0 +1,25 @@
+'use client'
+
+import { useLessonUnlocked } from 'hooks'
+import { useRouter } from 'next/navigation'
+import { useEffect, useState } from 'react'
+
+export default function LessonLocked({ children }) {
+  const router = useRouter()
+  const result = useLessonUnlocked()
+  const [lessonContent, setLessonContent] = useState(
+    <div className="flex h-full w-full grow flex-col items-center justify-center">
+      <span className="mb-10 text-4xl text-white">Loading Challenge...</span>
+    </div>
+  )
+
+  useEffect(() => {
+    if (typeof result === 'string') {
+      router.replace(result)
+    } else {
+      setLessonContent(children)
+    }
+  }, [])
+
+  return lessonContent
```

Comment:
Let's not put jsx into state, I think a better solution is something like this:
```jsx
export default function LessonLocked({ children }) {
  const router = useRouter()
  const result = useLessonUnlocked()

  useEffect(() => {
    if (typeof result === 'string') {
      router.replace(result)
    }
  }, [])

  return <div className="flex h-full w-full grow flex-col items-center justify-center">
    <span className="mb-10 text-4xl text-white">Loading Challenge...</span>
  </div>
```

Also I'm wondering why you're not just doing this in the `/app/[lang]/chapters/[slug]/[lesson]/layout.tsx` file since I doubt we'll reuse this new `LessonLocked.tsx` file much.